test/core/event_engine/posix/posix_event_engine_connect_test.cc

```diff
@@ -0,0 +1,240 @@
+                "Failed to connect to remote host: ", resolved_addr_str,
+                " with error: ",
+                absl::DeadlineExceededError("connect() timed out").ToString()));
+        absl::Status conn_reset_expected_status =
+            absl::CancelledError(absl::StrCat(
+                "Failed to connect to remote host: ", resolved_addr_str,
+                " with error: ",
+                absl::InternalError(absl::StrCat("getsockopt(SO_ERROR): ",
+                                                 std::strerror(ECONNRESET)))
+                    .ToString()));
+        // Most of the time, the attempt will fail with deadline exceeded error.
+        // Occasionally the kernel may send a RST to the connection attempt
+        // instead of simply dropping the SYN packet. The RST will cause
+        // the connection to fail before the timeout expires. The frequency
+        // threshold for the RST depends on the kernel version. We dont
+        // want to fail the test in both cases.
+        EXPECT_TRUE(status.status() == deadline_exceeded_expected_status ||
+                    status.status() == conn_reset_expected_status);
+        signal.Notify();
+      },
+      URIToResolvedAddress(target_addr), config,
+      memory_quota->CreateMemoryAllocator("conn-1"), 3s);
+  signal.WaitForNotification();
+  for (auto sock : *sockets) {
+    close(sock);
+  }
+  WaitForSingleOwner(std::move(posix_ee));
+}
+
+TEST(PosixEventEngineTest, IndefiniteConnectCancellationTest) {
+  std::string target_addr = absl::StrCat(
+      "ipv6:[::1]:", std::to_string(grpc_pick_unused_port_or_die()));
+  auto resolved_addr = URIToResolvedAddress(target_addr);
+  std::shared_ptr<EventEngine> posix_ee = std::make_shared<PosixEventEngine>();
+  std::string resolved_addr_str =
+      SockaddrToString(&resolved_addr, true).value();
+  auto sockets = CreateConnectedSockets(resolved_addr);
+  ASSERT_TRUE(sockets.ok());
+  grpc_core::ChannelArgs args;
+  auto quota = grpc_core::ResourceQuota::Default();
+  args = args.Set(GRPC_ARG_RESOURCE_QUOTA, quota);
+  ChannelArgsEndpointConfig config(args);
+  auto memory_quota = absl::make_unique<grpc_core::MemoryQuota>("bar");
+  auto connection_handle = posix_ee->Connect(
+      [](absl::StatusOr<std::unique_ptr<EventEngine::Endpoint>> /*status*/) {
+        FAIL() << "The on_connect callback should not have run since the "
+                  "connection attempt was cancelled.";
+      },
+      URIToResolvedAddress(target_addr), config,
+      memory_quota->CreateMemoryAllocator("conn-2"), 3s);
+  ASSERT_TRUE(posix_ee->CancelConnect(connection_handle));
+  for (auto sock : *sockets) {
+    close(sock);
+  }
+  WaitForSingleOwner(std::move(posix_ee));
+}
+
+}  // namespace posix_engine
+}  // namespace grpc_event_engine
+
+int main(int argc, char** argv) {
+  grpc::testing::TestEnvironment env(&argc, argv);
+  ::testing::InitGoogleTest(&argc, argv);
+  if (!grpc_core::IsPosixEventEngineEnablePollingEnabled()) {
```

Comment:
I don't think these _tests_ should be skipped. The experiment determines whether or not to use the EventEngine in production code ... the engine tests themselves should always be enabled. If some tests need to be skipped because the implementation is not ready to be tested, that's an important distinction to make.

Reply:
Per offline discussion I think we can keep this temporarily. The experiment flag is enabled in asan/tsan and debug builds. Its only disabled in opt builds. If the flag is not set, event engine doesn't create a poller and thus these tests would crash.